### PR TITLE
arch/arm64/arm64_mmu: add mmu_nxrt_config

### PR DESCRIPTION
**Pull request by HidenoriMatsubayashi**
## Summary
This change is a refactoring to match the implementation of mmu_regions which is user provided platform regions.

## Impact
None

## Testing
None

Signed-off-by: Hidenori Matsubayashi <hidenori.matsubayashi@gmail.com>

### CODE REVIEW

**Review by xiaoxiang781216 — APPROVED** [review]

File: AUTHORS [hunk 1]
```diff
@@ -78,6 +78,7 @@ Vasilijev Alexand Anatoljevich (Alexander Vasiliev)
 Wolfgang Gerd Reißnegger (Wolfgang Reißnegger)
 Xiang Xiao
 Yuuichi Nakamura
+Hidenori Matsubayashi
```
Comment (xiaoxiang781216) [inline AUTHORS:81]: move after line 29

Comment (HidenoriMatsubayashi) [reply AUTHORS]: Thanks for review. Fixed.

**Comment by masayuki2009** [comment]
@HidenoriMatsubayashi 
Please fix up the second commit. We don't need the commit message.

**Comment by masayuki2009** [comment]
@HidenoriMatsubayashi 
By the way, did you sign the ICLA?
https://www.apache.org/licenses/contributor-agreements.html
https://www.apache.org/licenses/icla.pdf

**Comment by HidenoriMatsubayashi** [comment]
@masayuki2009 

> Please fix up the second commit. We don't need the commit message.

What do you mean? Should I use "git commit --amend"?

> By the way, did you sign the ICLA?

Not yet, but it seems to be a troublesome process...I'll revert the change of AUTHORS.

**Comment by HidenoriMatsubayashi** [comment]
Thanks. IC. will do that.

**Comment by masayuki2009** [comment]
@HidenoriMatsubayashi 
In your case, you can not use `git commit --amend` which just amends the last commit.
Instead, you can use `git rebase -i HEAD~3` and drop the 3rd commit (i.e. Update AUTHORS) and fix up the second commit which will result in only the one commit.

**Comment by HidenoriMatsubayashi** [comment]
Done. Can you please approve / review / merge?

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by masayuki2009** [comment]
@HidenoriMatsubayashi 
I will merge this PR after the CI completes.

**Review by masayuki2009 — APPROVED** [review]

**Pull request merged**